app-ios/Sources/SearchFeature/SearchView.swift

```diff
@@ -104,6 +157,83 @@ public let searchReducer = Reducer<SearchState, SearchAction, SearchEnvironment>
             return .none
         case .session:
             return .none
+        case .selectCategory(let category):
+            state.filterCategories.append(category)
+            return .none
+        case .deselectCategory(let category):
+            let index = state.filterCategories.firstIndex(of: category)
+            guard let index = index else { return .none }
```

Comment:
This can be combined 🙏🏼 
```suggestion
            guard let index = state.filterCategories.firstIndex(of: category) else { return .none }
```